tests/test_parallelreduce.py

```diff
@@ -40,5 +44,53 @@ def test_add_squares(self):
         self.assertEqual(expected_result, result)
 
 
+@pk.workload
+class SquareSumFloat:
+    def __init__(self, n):
+        self.N: int = n
+        self.total: pk.double = 0
+
+    @pk.main
+    def run(self):
+        self.total = pk.parallel_reduce(self.N, self.squaresum)
+
+    @pk.workunit
+    def squaresum(self, i: float, acc: pk.Acc[pk.double]):
+        acc += i * i
+
+
+@pk.workload
+class SquareSumInt:
+    def __init__(self, n):
+        self.N: int = n
+        self.total: pk.int64 = 0
+
+    @pk.main
+    def run(self):
+        self.total = pk.parallel_reduce(self.N, self.squaresum)
+
+    @pk.workunit
+    def squaresum(self, i: int, acc: pk.Acc[pk.int64]):
```

Comment:
Locally I see only `1` test failure, for the largest integer value parametrized in the new test below. It appears to me that `i` is a 32-bit integer here, and that is causing overflow. Of course, I tried changing `i` to `pk.int64`, but this doesn't seem to translate or compile.

So, there appear to be a few different issues when attempting to match the NumPy reduction behavior here.